clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -473,6 +473,15 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                             memref.marker.marker_value == shard->stream->get_version(),
                         "Stream interface version != trace marker");
     }
+    if (memref.marker.type == TRACE_TYPE_MARKER &&
+        memref.marker.marker_type == TRACE_MARKER_TYPE_CHUNK_FOOTER &&
+        !shard->skipped_instrs_) {
```

Comment:
I see that tool.drcacheoff.skip failed with `Trace invariant failure in T1992554 at ref # 151 (1 instrs since timestamp 13352268558646620): Chunks do not increase monotonically`, which is why this relaxation for skipped_instrs_ was added. But couldn't we still check `cur GT last` in that case?

Reply:
Yes, we could.  Probably the message should be different for that case.  And it needs new tests: but the unit test framework doesn't look like it has any mechanism to do a skip.  So this will take some time.

Reply:
I made a mock skip flag but skipped the serial test as it's more complicated.  PTAL.